### ISSUE TITLE
Feature Request: Component for <i>

### ISSUE THREAD

**Issue by amahdy** [vaadin/flow#5829]
I want something more formal and not use `"<i>" + italicBoy + "</i>"` .. something like `new H1()`.

**Comment by Legioth** [comment]
This also raises a question about all the other common elements that have a purely representational function, e.g. `<b>` and `<u>` as well as semantic counterparts such as `<em>` and `<strong>`.

**Comment by amahdy** [comment]
I would really love to have a way to style a text, no need for a component.

`add(new H1("Hello " + TextStyler.setStrong(username) + "!"));`

**Comment by Legioth** [comment]
> `add(new H1("Hello " + TextStyler.setStrong(username) + "!"));`

This would be a huge security problem since it would mean that the string passed to the `H1` constructor would have to be interpreted as HTML instead of plain text.

**Comment by heruan** [comment]
I agree that passing HTML isn't safe, but in general it would be useful to have a safe way to place components in text templates (especially for I18N, where position may change), e.g.

```java
var text = getTranslation("my-key");
var template = new TextTemplate(text, new Italic(username));
add(template);
```

where `text` could be:
- English: `Hello, $0!`
- Italian: `Ciao, $0!`

**Comment by amahdy** [comment]
> This would be a huge security problem since it would mean that the string passed to the `H1` constructor would have to be interpreted as HTML instead of plain text.

How about something like old school label, accept HTML with the developer's responsibility, and takes text by default.

Also how about:

`add(new H1("Hello ", TextStyler.setStrong(username), "!"));`

So `TextStyler` can be interpreted differently and executed later.

The `TextTemplate` idea proposed looks good. Need to think about all possible combinations.

**Comment by Legioth** [comment]
`new H1("Hello ", TextStyler.setStrong(username), "!")` has the problem of the varags type when mixing `String` and some other type (unless abusing `CharSequence` or `Comparable<String>` which would be semantically weird). It basically only leaves `Object ... content` as the parameter type which instead causes other problems.

The template approach is then more promising. There's also a similar concept in the form of `SafeHtml` in GWT, though it's slightly convoluted because of the way it's designed to do as much as possible already at compile time.
